symfony/plugins/orangehrmAdminPlugin/Api/EmailConfigurationAPI.php

```diff
@@ -0,0 +1,366 @@
+        }
+        return $this->emailConfigurationService;
+    }
+
+    /**
+     * @param EmailConfigurationService $emailConfigurationService
+     */
+    public function setEmailConfigurationService(EmailConfigurationService $emailConfigurationService): void
+    {
+        $this->emailConfigurationService = $emailConfigurationService;
+    }
+
+    /**
+     * @return EndpointResourceResult
+     * @throws Exception
+     */
+    public function getOne(): EndpointResourceResult
+    {
+        $emailConfiguration = $this->getEmailConfigurationService()->getEmailConfigurationDao()->getEmailConfiguration(
+        );
+        if (!$emailConfiguration instanceof EmailConfiguration) {
+            $emailConfiguration = new EmailConfiguration();
+            $emailConfiguration->setMailType("");
+            $emailConfiguration->setSentAs("");
+            $emailConfiguration->setSmtpHost("");
+            $emailConfiguration->setSmtpPort(null);
+            $emailConfiguration->setSmtpUsername("");
+            $emailConfiguration->setSmtpPassword("");
+            $emailConfiguration->setSmtpAuthType("");
+            $emailConfiguration->setSmtpSecurityType("");
+        }
+
+        return new EndpointResourceResult(EmailConfigurationModel::class, $emailConfiguration);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID
+            ),
+        );
+    }
+
+    /**
+     * @return EndpointCollectionResult
+     * @throws NotImplementedException
+     */
+    public function getAll(): EndpointCollectionResult
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws NotImplementedException
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws NotImplementedException
+     */
+    public function create(): EndpointResourceResult
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws NotImplementedException
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     * @throws Exception
+     */
+    public function update(): EndpointResourceResult
+    {
+        $emailConfiguration = $this->saveEmailConfigurationInfo();
+        $testEmail = $this->getRequestParams()->getString(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_TEST_EMAIL_ADDRESS
+        );
+        $testEmailStatus = 1;
+        if (!empty($testEmail)) {
+            try {
+                $this->getEmailConfigurationService()->sendTestMail($testEmail);
+            } catch (Exception $e){
```

Comment:
don't we need to catch specific exception

Reply:
yes, set it as Swift_TransportException